test/integration-tests/Uniswap.test.ts

```diff
@@ -1,235 +1,422 @@
+      let amountAMin = amountADesired;
+      let amountBMin = amountBDesired;
+      let to = alice.address;
+      let deadline = Math.round((new Date()).getTime() / 1000) + 86400;
+      await (await usdcContract.connect(alice).mint(alice.address, amountADesired)).wait();
+      await (await wethContract.connect(alice).deposit({ value: amountBDesired })).wait();
+
+      await (await usdcContract.connect(alice).approve(uniswapV2Router.address, MAX_UINT)).wait();
+      await (await wethContract.connect(alice).approve(uniswapV2Router.address, MAX_UINT)).wait();
+
+      await (await uniswapV2Router.connect(alice).addLiquidity(
+        tokenA,
+        tokenB,
+        amountADesired,
+        amountBDesired,
+        amountAMin,
+        amountBMin,
+        to,
+        deadline
+      )).wait();
+
+      let reserves = await uniswapV2Router.getReserves(tokenA, tokenB);
+
+      expect(reserves[0]).to.be.eq(amountADesired);
+      expect(reserves[1]).to.be.eq(amountBDesired);
+    })
+
+    it.skip('swap dai to weth', async () => {
+
+      const amountInDAI = expandTo18DecimalsBN(100);
+
+      let swapper = bob;
+      let swapperAddress = bob.address;
+
+      await daiContract.connect(swapper).approve(uniswapV2Router.address, MAX_UINT);
+      await wethContract.connect(swapper).approve(uniswapV2Router.address, MAX_UINT);
+
+      let amountIn = amountInDAI;
+      let amountOutMin = 0;
+      let path = [daiContract.address, wethContract.address];
+      let to = swapperAddress;
+      let deadlile = Math.round((new Date()).getTime() / 1000) + 86400;
+      await (await uniswapV2Router.connect(bob).swapExactTokensForTokens(
+        amountIn,
+        amountOutMin,
+        path,
+        to,
+        deadlile
+      )).wait();
+
     })
-    alice = await ethers.getSigner(ALICE_ADDRESS)
-    bob = (await ethers.getSigners())[1]
-    daiContract = new ethers.Contract(DAI.address, TOKEN_ABI, bob)
-    wethContract = new ethers.Contract(WETH.address, TOKEN_ABI, bob)
-    usdcContract = new ethers.Contract(USDC.address, TOKEN_ABI, bob)
-    permit2 = (await deployPermit2()).connect(bob) as Permit2
-    router = (await deployUniversalRouter(permit2)).connect(bob) as UniversalRouter
-    planner = new RoutePlanner()
-
-    // alice gives bob some tokens
-    await daiContract.connect(alice).transfer(bob.address, expandTo18DecimalsBN(100000))
-    await wethContract.connect(alice).transfer(bob.address, expandTo18DecimalsBN(100))
-
-    // Bob max-approves the permit2 contract to access his DAI and WETH
-    await daiContract.connect(bob).approve(permit2.address, MAX_UINT)
-    await wethContract.connect(bob).approve(permit2.address, MAX_UINT)
   })
 
   describe('Trade on Uniswap with Permit2, giving approval every time', () => {
     describe('ERC20 --> ERC20', () => {
-      let permit: PermitSingle
+      let permitSingle: PermitSingle;
+      beforeEach(async () => {
+        planner = new RoutePlanner();
```

Comment:
Can it be removed?

Reply:
No, because it is necessary for test logic